src/mesh/analysis.rs

```diff
@@ -1213,31 +1368,100 @@ mod tests {
             ],
         ];
 
-        let computed_loops = border_edge_loops(&edge_sharing_map);
+        if let BorderEdgeLoopsResult::Found(computed_loops) = border_edge_loops(&edge_sharing_map) {
+            assert_eq!(computed_loops.len(), 2);
+            assert!(
+                computed_loops[0].len() == correct_loops[0].len()
+                    || computed_loops[0].len() == correct_loops[1].len()
+            );
+
+            if computed_loops[0].len() == correct_loops[0].len() {
+                assert_eq!(computed_loops[1].len(), correct_loops[1].len());
+                for (i, correct_loop) in correct_loops.iter().enumerate() {
+                    for edge in correct_loop {
+                        assert!(computed_loops[i].iter().any(|e| e == edge));
+                    }
+                }
+            } else {
+                assert_eq!(computed_loops[1].len(), correct_loops[0].len());
+                for (i, correct_loop) in correct_loops.iter().enumerate() {
+                    for edge in correct_loop {
+                        assert!(computed_loops[(i + 1) % 2].iter().any(|e| e == edge));
+                    }
+                }
+            }
```

Comment:
Isn't this a bit too generic? We know there are exactly two loops. Should probably extract them to variables and do comparisons on those instead of obfuscating with `[(i + 1) % 2]`, especially if the expression just means "the other of the two loops".